plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java

```diff
@@ -196,50 +197,47 @@ public void invalidateCaches(String tableLocation)
 
     public List<AddFileEntry> getActiveFiles(TableSnapshot tableSnapshot, ConnectorSession session)
     {
-        String tableLocation = tableSnapshot.getTableLocation().toString();
-        DeltaLakeDataFileCacheEntry cachedTable = activeDataFileCache.getIfPresent(tableLocation);
-
-        if (cachedTable == null || cachedTable.getVersion() > tableSnapshot.getVersion()) {
+        try {
+            String tableLocation = tableSnapshot.getTableLocation().toString();
             FileSystem fileSystem = getFileSystem(tableSnapshot, session);
-            try (Stream<AddFileEntry> entries = getEntries(
-                    tableSnapshot,
-                    ImmutableSet.of(ADD),
-                    this::activeAddEntries,
-                    session,
-                    fileSystem,
-                    hdfsEnvironment,
-                    fileFormatDataSourceStats)) {
-                List<AddFileEntry> activeFiles = entries.collect(toImmutableList());
-                if (cachedTable == null) {
-                    DeltaLakeDataFileCacheEntry cache = new DeltaLakeDataFileCacheEntry(tableSnapshot.getVersion(), activeFiles);
-                    activeDataFileCache.put(tableLocation, cache);
-                }
-                else {
-                    log.warn("Query run with outdated Transaction Log Snapshot, retrieved stale table entries for table: %s and query %s",
```

Comment:
restore

Reply:
Restoring this is not that simple.  I would need to check if data was loaded from the cache or file system. I would need to add atomic boolean etc (in similar fashion to what Marius suggested). Do you think it makes sense to add all this for a simple warn to be logged ?

Reply:
I think it belongs to 
```
if (cachedTable.getVersion() > tableSnapshot.getVersion()) {
                return loadActiveFiles(tableSnapshot, session, fileSystem)
```

block? Please double check me on this

Reply:
you are right, thanks